### PR TITLE
Compile ast fix

### PR DESCRIPTION
**Pull request by talsaiag**
**Fixed issue:** http://ironpython.codeplex.com/workitem/35707
**Title:** builtin’s compile function inconsistent with CPython
Allowing retrieving the file name and line (and col) number of an ast node complied using the builtin’s compile function.

### CODE REVIEW

**Comment by slide** [comment]
Can you squash this?

File: Languages/IronPython/IronPython/Compiler/Ast/DelStatement.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public class DelStatement : Statement {
                 statements.Add(_expressions[i].TransformDelete());
             }
             statements.Add(AstUtils.Empty());
-            return GlobalParent.AddDebugInfo(MSAst.Expression.Block(statements), Span);
+            return /*AppendLine*/(GlobalParent.AddDebugInfo(MSAst.Expression.Block(statements), Span));
```
Comment (slide) [inline Languages/IronPython/IronPython/Compiler/Ast/DelStatement.cs]: If this is commented, just remove it

Comment (talsaiag) [inline Languages/IronPython/IronPython/Compiler/Ast/DelStatement.cs]: I will remove all of them as requested in a sec.
(I initially added the commented out function calls to problematic places - as a place holder, to tell that they are not supposed to be there and it's not a mistake.)

**Comment by slide** [comment]
It would really be better if you squashed the commits on this.

**Comment by talsaiag** [comment]
Squashed irrelevant fix+merge (fix already implemented and merged elsewhere: pull request #1285 )

**Comment by rassilon** [comment]
I've pushed a squashed version of this PR to [rassilon:squashed_compile_ast_fix](https://github.com/rassilon/main/tree/squashed_compile_ast_fix)

File: Languages/IronPython/IronPython/Modules/_ast.cs [hunk 2]
```diff
@@ -876,6 +887,11 @@ public abstract class @operator : AST
         [PythonType]
         public abstract class slice : AST
         {
+            public AstExpression AttachLoc(AstExpression other)
```
Comment (rassilon) [inline Languages/IronPython/IronPython/Modules/_ast.cs]: Shouldn't the accessibility here be internal?

Comment (talsaiag) [inline Languages/IronPython/IronPython/Modules/_ast.cs]: actually yeah, would you like me to change it in my branch?
and should i squash the whole PR to a single commit?

File: Languages/IronPython/IronPython/Modules/_ast.cs [hunk 3]
```diff
@@ -1381,13 +1403,15 @@ internal ClassDef(ClassDefinition def)
                         _decorator_list.Add(Convert(expr));
                 } else
                     _decorator_list = PythonOps.MakeEmptyList(0);
-            }
+                _headerIndex = def.HeaderIndex;
+                }
```
Comment (rassilon) [inline Languages/IronPython/IronPython/Modules/_ast.cs]: This brace needs to be shifted left.

File: Languages/IronPython/IronPython/Modules/_ast.cs (hunk 2 above)
Comment (rassilon) [inline Languages/IronPython/IronPython/Modules/_ast.cs]: That seemed to be what slide was asking for.

**Comment by rassilon** [comment]
I've made the above changes to my squashed branch, resquashed and force-pushed.

File: Languages/IronPython/IronPython/Modules/_ast.cs (hunk 2 above)
Comment (talsaiag) [inline Languages/IronPython/IronPython/Modules/_ast.cs]: There were three additional commits, I think he made his comment regarding:
First, fixing #1285 in my branch as part of my code which caused a spacing conflict.
A second and third, merging from the upstream main to mine, and from my main to this branch after the fix was introduced there.
I just removed those (the fix is already on the main branch, there are no conflicts with current state and I removed the last unintended extra 3 commits)

**Comment by simplicbe** [comment]
@rassilon I would like to merge this as soon as possible. What tests have you ran agains the changes?

**Comment by simplicbe** [comment]
I've cloned, compiled and tested everything using `test-ipy.cmd /all` for the current main branch and the branch which should be merged. The results are identically. So i assume everything should be fine.

**Pull request merged**